r-session-04.qmd

```diff
@@ -0,0 +1,800 @@
+---
+subtitle: "Stochastic SIR models"
+abstract-title: ""
+abstract: |
+    *review of basic algorithms for stochastic epidemic models*
+execute:
+    warning: false
+metadata-files:
+    - metadata/matthewferrari.yml
+    - metadata/mathjax-packages.yml
+editor:
+  markdown:
+    wrap: sentence
+---
+
+# R Session 03
+
+## Setup
+
+```{r}
+library(here)
+library(rio)
+library(deSolve)
+library(tidyverse)
+library(ggtext)
+library(gt)
+```
+
+```{r}
+theme_set(theme_minimal())
+```
+
+
+
+## What is stochasticity and where does it come from?
+
+Much of the world is uncertain (i.e. we don't know exactly how things work, what values are, or what tomorrow will hold).
+Some of that uncertainty is, at least theoretically, knowable and some is not. 
+For example, in our discussion of estimating $R_0$, there may be a very real $R_0$ for a given population and pathogen, even if we don't know it. 
+Thus, our estimate of $R_0$ may be "uncertain" (e.g. has a confidence interval around it, reflecting our certainty), but the models we've been developing so far are "deterministic", so conditional on a given value of $R_0$ the resulting epidemic curve is exactly specified by the model.
+If we return to the code from R-session 1, we can plot a single deterministic realization of a model with $R_0 = 1.8$.
+
+```{r}
+#| echo: true
+#| message: false
+#| out-width: 100%
+#| column: body
+library(tidyverse)
+library(deSolve)
+
+theme_set(theme_minimal())
+
+sir_model <- function(time, state, params, ...) {
+  transmission <- params["transmission"]
+  recovery <- 1 / params["duration"]
+
+  S <- state["S"]
+  I <- state["I"]
+  R <- state["R"]
+
+  dSdt <- -transmission * S * I
+  dIdt <- (transmission * S * I) - (recovery * I)
+  dRdt <- recovery * I
+
+  return(list(c(dSdt, dIdt, dRdt)))
+}
```

Comment:
Might make sense to separate these sections into distinct code chunks that focus on a single thing e.g., the model function definition, setting up params and running the model, converting to df, then plotting